src/EnergyPlus/WeatherManager.cc

```diff
@@ -154,59 +148,61 @@ namespace WeatherManager {
+
+        constexpr std::string_view routineName = "CheckIfAnyUnderwaterBoundaries";
+        
+        auto const &ipsc = state.dataIPShortCut;
+        ipsc->cCurrentModuleObject = "SurfaceProperty:Underwater";
+        int Num = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, ipsc->cCurrentModuleObject);
         for (int i = 1; i <= Num; i++) {
             state.dataInputProcessing->inputProcessor->getObjectItem(state,
-                                                                     state.dataIPShortCut->cCurrentModuleObject,
+                                                                     ipsc->cCurrentModuleObject,
                                                                      i,
-                                                                     state.dataIPShortCut->cAlphaArgs,
+                                                                     ipsc->cAlphaArgs,
                                                                      NumAlpha,
-                                                                     state.dataIPShortCut->rNumericArgs,
+                                                                     ipsc->rNumericArgs,
                                                                      NumNumber,
                                                                      IOStat,
-                                                                     state.dataIPShortCut->lNumericFieldBlanks,
-                                                                     state.dataIPShortCut->lAlphaFieldBlanks,
-                                                                     state.dataIPShortCut->cAlphaFieldNames,
-                                                                     state.dataIPShortCut->cNumericFieldNames);
-            state.dataWeatherManager->underwaterBoundaries.emplace_back();
-            auto &underwaterBoundary = state.dataWeatherManager->underwaterBoundaries[i - 1];
-            underwaterBoundary.Name = state.dataIPShortCut->cAlphaArgs(1);
-            underwaterBoundary.distanceFromLeadingEdge = state.dataIPShortCut->rNumericArgs(1);
-            underwaterBoundary.OSCMIndex = UtilityRoutines::FindItemInList(underwaterBoundary.Name, state.dataSurface->OSCM);
+                                                                     ipsc->lNumericFieldBlanks,
+                                                                     ipsc->lAlphaFieldBlanks,
+                                                                     ipsc->cAlphaFieldNames,
+                                                                     ipsc->cNumericFieldNames);
+            state.dataWeather->underwaterBoundaries.emplace_back();
+            auto &underwaterBoundary = state.dataWeather->underwaterBoundaries[i - 1];
+            underwaterBoundary.Name = ipsc->cAlphaArgs(1);
+
+            ErrorObjectHeader eoh{routineName, ipsc->cCurrentModuleObject, underwaterBoundary.Name};
```

Comment:
Use these wrappers for common errors. See `UtilityRoutines` for available wrappers.